src/ansi-c/ansi_c_internal_additions.cpp

```diff
@@ -120,6 +122,25 @@ static std::string architecture_string(T value, const char *s)
          std::string(s) + "=" + std::to_string(value) + ";\n";
 }
 
+using max_alloc_sizet = uint64_t;
+/// The maximum allocation size is determined by the number of bits that
+/// are left in the pointer of width \p pointer_width after allowing for the
+/// signed bit, and the bits used for the objects ID (determined by
+/// \p object_bits).
+/// \param pointer_width: The width of the pointer
+/// \param object_bits : The number of bits used to represent the ID
+/// \return The size in bytes of the maximum allocation supported.
+static max_alloc_sizet
+max_malloc_size(std::size_t pointer_width, std::size_t object_bits)
+{
+  PRECONDITION(pointer_width >= 1);
+  PRECONDITION(object_bits < pointer_width - 1);
```

Comment:
I think this should be `object_bits < pointer_width`. Just one bit for the offset would still be ok.

Reply:
I think not: you need one bit for the sign? 
Pointer width = 8
Object bits = 7
1-1 = 0 bits for the offset?

Reply:
Just saw second comment - OK I misunderstood what you said - I thought the signed bit was used for something else, rather than the offset?

Reply:
The offset is signed. So with just one bit, 0 (in binary) would be 0 (in decimal), and 1 (in binary) would be -1 (in decimal).